core/src/main/java/org/infinispan/interceptors/impl/TxInterceptor.java

```diff
@@ -630,6 +630,7 @@ protected Object getKey(CacheEntry<K, V> value) {
       private final TxInvocationContext<LocalTransaction> ctx;
       // We store all the not yet seen context entries here.  We rely on the fact that the cache entry reference is updated
       // if a change occurs in between iterations to see updates.
+      // TODO Dan: Only true with REPEATABLE_READ isolation level, not true with READ_COMMITTED
```

Comment:
what is the point here? with RC, that change happens in the context as well.

Reply:
Not always. With RC, a read will store the `InternalCacheEntry` in the context as is. If the same tx modifies the entry, a new `ReadCommittedEntry` is created and stored in the context instead -- the old ICE is not modified until commit (and even then, only if the entry is local).

Reply:
well, RC doesn't work if the entry is remote. I'm planning to fix it by removing the entry from the context if it isn't modified. that way, it would force to read it from DataContainer/CacheLoader/Remote node. 

so, should we expose some kind of iterator in `EntryLookup` instead of using a snapshot?

Reply:
@pruivo I'm not sure what you mean, this queue is maintained by `TxInterceptor` and not by `EntryLookup` implementations.

Reply:
we can discuss this later. I triggered another CI run and lets integrate this :)